tgui/packages/tgui/interfaces/MatrixMathTester.js

```diff
@@ -0,0 +1,124 @@
+            <Stack.Item grow>
+              <Stack fill>
+                <Stack.Item grow>
+                  <NumberInput
+                    value={matrix_a}
+                    step={0.005}
+                    format={value => toFixed(value, 2)}
+                    width="70px"
+                    onChange={(e, value) => act('change_var', {
+                      var_name: "a",
+                      var_value: value,
+                    })} />
+                </Stack.Item>
+                <Stack.Item grow>
+                  <Box>
+                    <NumberInput
+                      value={matrix_d}
+                      step={0.005}
+                      format={value => toFixed(value, 2)}
+                      width="70px"
+                      onChange={(e, value) => act('change_var', {
+                        var_name: "d",
+                        var_value: value,
+                      })} />
+                  </Box>
+                </Stack.Item>
+                <Stack.Item grow>
+                  <Input
+                    backgroundColor="red"
+                    placeholder="0 (fixed value)" />
+                </Stack.Item>
+              </Stack>
+            </Stack.Item>
+            <Stack.Item grow>
+              <Stack fill>
+                <Stack.Item grow>
+                  <NumberInput
+                    value={matrix_b}
+                    step={0.005}
+                    format={value => toFixed(value, 2)}
+                    width="70px"
+                    onChange={(e, value) => act('change_var', {
+                      var_name: "b",
+                      var_value: value,
+                    })} />
+                </Stack.Item>
+                <Stack.Item grow>
+                  <NumberInput
+                    value={matrix_e}
+                    step={0.005}
+                    format={value => toFixed(value, 2)}
+                    width="70px"
+                    onChange={(e, value) => act('change_var', {
+                      var_name: "e",
+                      var_value: value,
+                    })} />
+                </Stack.Item>
+                <Stack.Item grow>
+                  <Input
+                    backgroundColor="red"
+                    placeholder="0 (fixed value)" />
+                </Stack.Item>
+              </Stack>
+            </Stack.Item>
+            <Stack.Item grow>
+              <Stack fill>
+                <Stack.Item grow>
+                  <NumberInput
+                    value={matrix_c}
+                    step={0.005}
+                    format={value => toFixed(value, 2)}
+                    width="70px"
+                    onChange={(e, value) => act('change_var', {
+                      var_name: "c",
+                      var_value: value,
+                    })} />
+                </Stack.Item>
+                <Stack.Item grow>
+                  <NumberInput
```

Comment:
You repeat this a lot, I think you could probably make it a small component.

Reply:
*soft sobbing noises*

Reply:
don't do this to me...................

Reply:
oh wait that's not that hard

Reply:
dumby